src/Corax/Queries/BinaryMatch.cs

```diff
@@ -46,6 +47,53 @@ public int Fill(Span<long> buffer)
         public int AndWith(Span<long> buffer)
         {
             return _andWith(ref this, buffer);
-        }        
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void Score(Span<long> matches, Span<float> scores)
+        {
+            // Nothing to do if there is no boosting happening at this level. 
+            bool innerBoosting = _inner.IsBoosting;
+            bool outerBoosting = _outer.IsBoosting;
+            if (innerBoosting == false && outerBoosting == false)
+                return;
+            
+            // From now on we have boosting happening somewhere in this chain. 
+
+            // If there are two chains we need to combine them.
+            if (innerBoosting == true && outerBoosting == true)
+            {
+                _inner.Score(matches, scores);
+                
+                // TODO: We are not going to use stackalloc here.
+                Span<float> outerScores = stackalloc float[scores.Length];
+                outerScores.Fill(1); // We will fill the scores with 1.0
+
+                // We get the score for the outer chain.
+                _outer.Score(matches, outerScores);
+
+                // We multiply the scores from the outer chain with the current scores and return.
+                for(int i = 0; i < scores.Length; i++)
+                    scores[i] *= outerScores[i];
+                return;
+            }
+
+            // From now on, only a single requires score calculations. 
+
+            if (innerBoosting == true)
+            {
+                // Inner can still be not boosting. In this case it is, so we delegate
+                // the call into the boosting layer that provides us the information.
```

Comment:
I don't understand this comment

Reply:
You can have boosting on one side of the binary operator that it is bubbling up, but not on the other. So you can optimize for each one of those cases.